core/kernel/lockdep.c

```diff
@@ -0,0 +1,397 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2018, Linaro Limited
+ */
+
+#include <assert.h>
+#include <kernel/lockdep.h>
+#include <kernel/unwind.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/queue.h>
+#include <util.h>
+
+/* lockdep_node::flags values */
+/* Flags used for depth-first topological sorting */
+#define LOCKDEP_NODE_TEMP_MARK		BIT(0)
+#define LOCKDEP_NODE_PERM_MARK		BIT(1)
+/* Flag used during breadth-first search (print shortest cycle) */
+#define LOCKDEP_NODE_BFS_VISITED	BIT(2)
+
+/* Find node in graph or add it */
+static struct lockdep_node *lockdep_add_to_graph(
+				struct lockdep_node_head *graph,
+				uintptr_t lock_id)
+{
+	struct lockdep_node *node = NULL;
+
+	TAILQ_FOREACH(node, graph, link)
+		if (node->lock_id == lock_id)
+			return node;
+
+	node = calloc(1, sizeof(*node));
+	if (!node)
+		return NULL;
+
+	node->lock_id = lock_id;
+	STAILQ_INIT(&node->edges);
+	TAILQ_INSERT_TAIL(graph, node, link);
+
+	return node;
+}
+
+static vaddr_t *dup_call_stack(vaddr_t *stack)
+{
+	int n = 0;
+	vaddr_t *nstack = NULL;
+
+	if (!stack)
+		return NULL;
+	while (stack[n])
+		n++;
+	nstack = malloc(n * sizeof(vaddr_t));
+	if (!nstack)
+		return NULL;
+	memcpy(nstack, stack, n * sizeof(vaddr_t));
+	return nstack;
+}
+
+static void lockdep_print_call_stack(vaddr_t *stack)
+{
+	vaddr_t *p = NULL;
```

Comment:
min,or: useless init

Reply:
Yeah, it's a new rule, for security reasons we're now requiring that all local variables be initialized.
https://github.com/OP-TEE/optee_os/commit/439203cb04ce

Reply:
ok, thanks. I've removed my related comments.